src/inc/regdisp.h

```diff
@@ -370,7 +389,9 @@ inline void FillRegDisplay(const PREGDISPLAY pRD, PT_CONTEXT pctx, PT_CONTEXT pC
 
 #ifdef _TARGET_X86_
     pRD->pContext = pctx;
+#ifndef WIN64EXCEPTIONS
```

Comment:
This needs to be different. We need to use the common code that's below for _WIN64 for WIN64EXCEPTIONS on Linux x86 too. So please keep this part of the code that is for _TARGET_X86_ untouched, just change the condition from _TARGET_x86_  to #ifndef WIN64EXCEPTIONS and the #elif defined(_WIN64) below to #else. Then in the #ifdef _TARGET_AMD64_ etc add branch for _TARGET_X86_ and put the registers copying in there in a way similar to what we do for the other targets.
There is also an #ifdef for _TARGET_ARM below, which would need to be merged with the _WIN64 part. They are almost the same, so it would be good anyways. Just be careful, the ARM path has two additional tiny details,  explicit copying of Lr and Pc.